critwall/src/main/java/org/opencadc/critwall/FileSyncJob.java

```diff
@@ -241,41 +244,59 @@ public void run() {
      * @throws InterruptedException
      * @throws WriteException
      */
-    private StorageMetadata syncArtifact(Iterator<URL> urlIterator, URI artifact, StorageAdapter sa)
-        throws StorageEngageException, InterruptedException, WriteException {
+    private StorageMetadata syncArtifact(Iterator<URL> urlIterator, URI artifact, StorageAdapter sa, ArtifactDAO artDao)
```

Comment:
I didn't notice this before, but it's confusing to pass member variables as args to methods: the method can use the member var directly. Further, these 3 are all final specifically so methods cannot accidentally set the values to something else (like null). Only the iterator is needed).